File: adivina.dart
//adivina el numero

import 'dart:math' as math;
import './utilidades.dart';

void main() {
  //declarar variables
  int secreto = math.Random().nextInt(10) + 1;
  int numero = 0;
  String alias = '';

  encabezado('Adivina el numero', 50);
  alias = leerTexto('Ingresa un alias', 'alias');
  linea(50);

  jugar(secreto: secreto, numero: numero, alias: alias);
}

void jugar({required secreto, required numero, required alias}) {
  int intentos = 1;
  while (numero != secreto) {
    if (intentos > 3) {
      linea(50);
      print(
        '${alias.toUpperCase()}, perdiste le numero secreto era el ${secreto}',
      );
      linea(50);
      break;
    }
    intentos++;
    numero = leerNumeroEntero('Ingresa tu numero: ');
    if (numero < secreto) {
      print('tu numero es menor que el secreto...');
    } else if (numero > secreto) {
      print('tu numero es mayor que el secreto...');
    } else {
      linea(50);
      print(' ${alias.toUpperCase()}, Felicidades ganaste...');
      linea(50);
    }
  }
}

File: java/project2/src/arsistema/Adivina.java
package arsistema;

import java.util.Random;

public class Adivina {
    public static Random random = new Random();

    public static void main(String[] args) {
        // declarar variables (atributos de la clase):
        int secreto = random.nextInt(20)+1;
        int numero = 0;
        String alias = " ";

        Util.Encabezado("Bienvenido al juego de Adivina el numero", 50);
        alias = Util.leerTexto("Ingresa tu alias", "alias");
        jugar(secreto, numero, alias);
    }
    public static void jugar (int secreto, int numero, String alias) {
        String mensaje = "";
        int intentos = 1;

        while (numero != secreto) {
            if (intentos == 4) {
                Util.linea(50);
                mensaje = "Perdiste el numero secreto era el: ";
                System.out.printf("%s, %s %d %n", alias.toUpperCase(), mensaje, secreto);
                Util.linea(50);
                break;
            }
            numero = Util.leerNumeroEntero("Ingresa tu numero");
            intentos++;

            if (numero < secreto) {
                mensaje = "tu numero es menor al secreto...";
                System.out.printf("%s, %s %n", alias.toUpperCase(), mensaje);
            } else if(numero > secreto){
                mensaje = "tu numero es mayor al secreto...";
                System.out.printf("%s, %s %n", alias.toUpperCase(), mensaje);
            } else{
                Util.linea(50);
                mensaje = "Felicitaciones ganaste...";
                System.out.printf("%s, %s %n", alias.toUpperCase(), mensaje);
                Util.linea(50);
            }
        }
        
    }
}
